staging/src/k8s.io/apiextensions-apiserver/pkg/cmd/server/start.go

```diff
@@ -91,7 +91,7 @@ func (o *CustomResourceDefinitionsServerOptions) Complete() error {
 	return nil
 }
 
-func (o CustomResourceDefinitionsServerOptions) Config() (*apiserver.Config, error) {
+func (o *CustomResourceDefinitionsServerOptions) Config() (*apiserver.Config, error) {
 	// TODO have a "real" external address
 	if err := o.RecommendedOptions.SecureServing.MaybeDefaultWithSelfSignedCerts("localhost", nil, []net.IP{net.ParseIP("127.0.0.1")}); err != nil {
```

Comment:
are there side-effect to the options in  MaybeDefaultWithSelfSignedCerts?

Reply:
no

Reply:
https://github.com/sttts/kubernetes/blob/6ba30f678c232793430a98770e7a851f1e814fd2/staging/src/k8s.io/apiserver/pkg/server/options/serving.go#L269 mutates

Reply:
I know.  `CustomResourceDefinitionsServerOptions` will be mutated by `Config`. in any case no matter poniter-receiver or non-pointer. Because `CustomResourceDefinitionsServerOptions.RecommendedOptions` is pointer and `RecommendedOptions.SecureServing` is also pointer type.